File: tests/com/sefford/brender/filters/NullFilterTest.java
package com.sefford.brender.filters;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertNull;
import static org.mockito.MockitoAnnotations.initMocks;

@RunWith(RobolectricTestRunner.class)
public class NullFilterTest {

    private NullFilter filter;

    @Before
    public void setUp() throws Exception {
        initMocks(this);

        filter = new NullFilter();
    }

    @Test
    public void testPerformFiltering() throws Exception {
        assertNull(filter.performFiltering(""));
    }

    @Test
    public void testPublishResults() throws Exception {
        filter.publishResults(null, null);
    }
}

File: modules/brender-core/src/test/kotlin/com/sefford/brender/filters/NullFilterTest.kt
/*
 * Copyright (C) 2015 Saúl Díaz
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.sefford.brender.filters

import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.MockitoAnnotations.initMocks
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class NullFilterTest {

    lateinit var filter: NullFilter

    @Before
    @Throws(Exception::class)
    fun setUp() {
        initMocks(this)

        filter = NullFilter()
    }

    @Test
    @Throws(Exception::class)
    fun testPerformFiltering() {
        filter.filter("")
    }

}
